resource-types/typescript-example-website-monitor/src/handlers.ts

```diff
@@ -0,0 +1,316 @@
+            const createResponse: Response = await fetch(`${apiEndpoint}/v3/monitors`, {
+                method: 'POST',
+                headers: { ...Resource.DEFAULT_HEADERS, 'Api-Key': apiKey },
+                body: JSON.stringify({
+                    name: model.name,
+                    uri: model.uri,
+                    type: model.kind,
+                    frequency: model.frequency,
+                    status: model.status,
+                    locations: model.locations,
+                    slaThreshold: model.slaThreshold
+                } as Monitor)
+            });
+            await this.checkResponse(createResponse, logger, request.logicalResourceIdentifier);
+
+            // Use address from location header to read newly created monitor
+            // https://docs.newrelic.com/docs/apis/synthetics-rest-api/monitor-examples/manage-synthetics-monitors-rest-api#get-specific-monitor
+            const locationUrl = createResponse.headers.get('location');
+            if (!locationUrl) {
+                throw new exceptions.NotFound(this.typeName, request.logicalResourceIdentifier);
+            }
+            const response: Response = await fetch(locationUrl, {
+                method: 'GET',
+                headers: { ...Resource.DEFAULT_HEADERS, 'Api-Key': apiKey }
+            });
+            const monitor: Monitor = await this.checkResponse(response, logger, request.logicalResourceIdentifier);
+            model.id = monitor.id;
+            // model.apiKey = null;
+
+            // Setting Status to success will signal to CloudFormation that the operation is complete
+            progress.status = OperationStatus.Success;
+        } catch(err) {
+            logger.log(err);
+            if (err instanceof exceptions.BaseHandlerException) {
+                throw err;
+            }
+            throw new exceptions.InternalFailure(err.message);
+        }
+        logger.log('progress', progress);
+        return progress;
+    }
+
+    /**
+     * CloudFormation invokes this handler when the resource is updated
+     * as part of a stack update operation.
+     *
+     * @param session Current AWS session passed through from caller
+     * @param request The request object for the provisioning request passed to the implementor
+     * @param callbackContext Custom context object to allow the passing through of additional
+     * state or metadata between subsequent retries
+     * @param logger Logger to proxy requests to default publishers
+     */
+    @handlerEvent(Action.Update)
+    public async update(
+        session: Optional<SessionProxy>,
+        request: ResourceHandlerRequest<ResourceModel>,
+        callbackContext: CallbackContext,
+        logger: LoggerProxy
+    ): Promise<ProgressEvent> {
+        logger.log('request', request);
+        const model = new ResourceModel(request.desiredResourceState);
+        const { id, name } = request.previousResourceState;
+
+        if (!model.id) {
```

Comment:
Just a general comment here.  I don't believe this should have to be done in the code. The service shouldn't be passing something to update that doesn't have a `Id` since its a `primaryIdentifier`

Reply:
I believe this is needed to pass the contract tests